frontend/gateway/forwarder/forward.go

```diff
@@ -247,19 +246,26 @@ func (c *bridgeClient) NewContainer(ctx context.Context, req client.NewContainer
 			}
 		}
 		ctrReq.Mounts = append(ctrReq.Mounts, gateway.Mount{
-			Dest:      m.Dest,
-			Selector:  m.Selector,
-			Readonly:  m.Readonly,
-			MountType: m.MountType,
 			WorkerRef: workerRef,
-			CacheOpt:  m.CacheOpt,
-			SecretOpt: m.SecretOpt,
-			SSHOpt:    m.SSHOpt,
+			Mount: &opspb.Mount{
+				Dest:      m.Dest,
+				Selector:  m.Selector,
+				Readonly:  m.Readonly,
+				MountType: m.MountType,
+				CacheOpt:  m.CacheOpt,
+				SecretOpt: m.SecretOpt,
+				SSHOpt:    m.SSHOpt,
+			},
 		})
 	}
 
+	w, err := c.workers.GetDefault()
```

Comment:
We can leave this to follow-up but this looks logically wrong to have worker resolving in this level. `GetDefault()` is a function could be replaced with a more complex worker resolve function in the future but logic like that shouldn't be in this package but figured out in `llbsolver`

Reply:
Sounds good. Since the `worker.Info` is just an interface, we can pass an implementation that returns the specific worker resolved by `llbsolver` in the future. Either way, the downstream consumers (like the gateway forwarder) needs access to a `worker.Worker` and `[]client.WorkerInfo`, so we can just rename `GetDefault` -> `GetResolvedWorker` or something.